File: in_class_ex/Comparison Sort Exercise/c++/quick_sort.cpp
#include <iostream>
#include <vector>

// Function to partition the array into two subarrays based on a pivot element
// Elements smaller than the pivot are on the left, and elements greater than
// the pivot are on the right.
int partition(std::vector<int>& arr, int low, int high) {
    int pivot = arr[high]; // Choose the rightmost element as the pivot
    int i = (low - 1); // Index of the smaller element

    //Put your code here
}

// Function to perform Quick Sort
void quickSort(std::vector<int>& arr, int low, int high) {
    if (low < high) {
        // Partition the array into two subarrays
        int pi = partition(arr, low, high);

        // Recursively sort the subarrays
        // Put your code here
    }
}

// Function to print an array
void printArray(const std::vector<int>& arr) {
    for (int num : arr) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
}

int main() {
    std::vector<int> arr = {10, 7, 8, 9, 1, 5};
    int n = arr.size();

    std::cout << "Original Array: ";
    printArray(arr);

    quickSort(arr, 0, n - 1);

    std::cout << "Sorted Array: ";
    printArray(arr);

    return 0;
}

File: in_class_ex/Comparison Sort Exercise/python/quick_sort.py
def partition(arr, low, high):
    pivot = arr[high]
    i = low - 1

    # Put your code here

    return 

def quick_sort(arr, low, high):
    if low < high:
        pi = partition(arr, low, high)
        # Put your code here

# Example usage:
arr = [10, 7, 8, 9, 1, 5]
print("Original Array:", arr)
quick_sort(arr, 0, len(arr) - 1)
print("Sorted Array:", arr)
